core/src/main/java/uk/gov/gchq/koryphe/tuple/ReflectiveTuple.java

```diff
@@ -78,28 +78,53 @@ public Object getRecord() {
 
     @Override
     public Object get(final String reference) {
+        requireNonNull(reference, "field reference is required");
+
         Object selection;
-        try {
-            selection = invokeMethodGet(record, reference);
-        } catch (final IllegalAccessException | NoSuchMethodException | InvocationTargetException ignored) {
+        final int index = reference.indexOf(".");
+        if (index > -1) {
```

Comment:
What are your thoughts on some friendly variable names to help visualise what is going on?
same for put.
```
        final boolean hasDot = index > -1;
        if (hasDot) {
            final String referencePart = reference.substring(0, index);
            selection = get(referencePart);
            final boolean additionalNesting = index + 1 < reference.length();
            if (additionalNesting) {
                final Tuple<String> selectionAsTuple = selection instanceof Tuple ? ((Tuple) selection) : new ReflectiveTuple(selection);
                final String nestedReference = reference.substring(index + 1, reference.length());
                selection = selectionAsTuple.get(nestedReference);
            }
        }
```